src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -445,6 +488,9 @@ impl Net {
                 }
 
                 error!("net: Could not parse an RX descriptor: {err}");
+                // Notify queue about ready frames. We need this
+                // to bring queue into up to date state.
+                self.rx_buffer.finish_frame(queue);
```

Comment:
Could you explain to me why we call `finish_frame` here?

Reply:
Added a better comment. But in short, because now we write to the used_ring separately from bumping queue, if we have pending packet, `add_used` call will mess everything up.

Reply:
So, I think this might be a problem. If there is a frame to "finish" here, that's because we have rate limited it. So, running `finish_frame` on it, doesn't seem right.

Reply:
we can have a check if there is a rate limited frame and if so, we do `undo_pop` and `break`. This way we will have to parse same broken chain 2 or more times though.

Reply:
I'm not sure what's best here. @roypat WDYT?

Reply:
The other approach is to add this broken chain into the queue same way we do with valid chains with `write_used_element` (we can use `rx_buffer.used_descriptors` to calculate offset.) and add 1 to the `rx_buffer.used_descriptors`.

Reply:
So, we would do:

```rust
self.write_used_element(self.rx_buffer.used_descriptors, index, 0);
self.rx_buffer_used_descriptors += 1;
```

right? I think this works.

Reply:
maybe worth making the above snippet a function in `RxBuffers`

Reply:
doesn't seem to be working. Things are massively failing in Buildkite

Reply:
now should be fixed